src/buildercore/fastly/vcl/main.vcl

```diff
@@ -11,8 +11,18 @@ sub vcl_recv {
 sub vcl_fetch {
   #FASTLY fetch
 
-  if ((beresp.status == 500 || beresp.status == 503) && req.restarts < 1 && (req.request == "GET" || req.request == "HEAD")) {
-    restart;
+  if (beresp.status >= 500 && beresp.status < 600) {
+    if (stale.exists) {
+      return(deliver_stale);
+    }
+
+    if (req.restarts < 1 && (req.request == "GET" || req.request == "HEAD")) {
+      unset req.http.Cookie;
+
+      restart;
+    }
+
+    error 503;
```

Comment:
not sure if this loses the original HTTP status code, turning all `50x` into `500`. Would fit with the proxy model, but hides the real error